test/schema.jl

```diff
@@ -1,10 +1,17 @@
 @testset "schemas" begin
-
+    import Base
     using StatsModels: schema, apply_schema, FullRank
 
-    f = @formula(y ~ 1 + a + b + c + b&c)
-    df = (y = rand(9), a = 1:9, b = rand(9), c = repeat(["d","e","f"], 3))
+    f = @formula(y ~ 1 + a + b + c + b & c)
+    y = rand(9)
+    b = rand(9)
+
+    df = (y = y, a = 1:9, b = b, c = repeat(["d", "e", "f"], 3))
     f = apply_schema(f, schema(f, df))
     @test f == apply_schema(f, schema(f, df))
 
+    df2 = (y = y, a = 1:9, b = b, c = repeat(["d", "e", "f"], 3))
+
+    @test schema(df) == schema(df2)
+    @test isequal(schema(df), schema(df2))
```

Comment:
this is a good start but you should also test as many of the "edge cases" you can think of as possible, since this will serve to make sure we don't break things in the future.  In particular, I'd suggest adding tests for 

* schema/terms with different named variables are not equal
* schema/terms with different invariants (mean/var/min/max, contrasts) are not equal
* schema are generated based on different tables with the same invariants (e.g., df1 with each column repeated twice) are equal

And whatever else you can dream up :)

Reply:
I understood the first two points. Can you explain the last one in some detail ?

Reply:
the schema shouldn't depend on the exact data that generated it, as long as the unique levels (for categorical) or mean/std/min/max (for continuous) are the same.  you check it right now by creating what is essentially a deep copy of the table, but it would be good to also check using a table that's not exactly the same size but has the same invariants, e.g. something like

`df2 = (y = [df1.y; df1.y], a = [1:9; 1:9], b = [b; b], c = [df.c; df.c])`